contrib/platform/src/com/sun/jna/platform/win32/WevtapiUtil.java

```diff
@@ -0,0 +1,224 @@
+     *                          message strings. You can also set this parameter to NULL if the event property that
+     *                          you are formatting is defined in the Winmeta.xml file (for example, if level is set
+     *                          to win:Error). In the latter case, the service uses the Winmeta provider as
+     *                          the formatting context and will format only those message strings that you reference
+     *                          in your event that are defined in the Winmeta.xml file.
+     * @param event             [in] A handle to an event. The Flags parameter specifies the message string in
+     *                          the event that you want to format. This parameter must be NULL if the Flags parameter
+     *                          is set to EvtFormatMessageId.
+     * @param messageId         [in] The resource identifier of the message string that you want to format.
+     *                          To get the resource identifier for a message string, call
+     *                          the {@link Wevtapi#EvtGetPublisherMetadataProperty} function. Set this parameter only
+     *                          if the Flags parameter is set to EvtFormatMessageId.
+     * @param valueCount        [in] The number of values in the Values parameter.
+     * @param values            [in] An array of insertion values to use when formatting the event's message string.
+     *                          Typically, you set this parameter to NULL and the function gets the insertion values
+     *                          from the event data itself. You would use this parameter to override the default
+     *                          behavior and supply the insertion values to use. For example, you might use this
+     *                          parameter if you wanted to resolve a SID to a principal name before inserting the value.
+     *                          <p>
+     *                          To override the insertion values, the Flags parameter must be set to
+     *                          {@link Winevt.EVT_FORMAT_MESSAGE_FLAGS#EvtFormatMessageEvent},
+     *                          {@link Winevt.EVT_FORMAT_MESSAGE_FLAGS#EvtFormatMessageXml}, or
+     *                          {@link Winevt.EVT_FORMAT_MESSAGE_FLAGS#EvtFormatMessageId}, If Flags is set to
+     *                          {@link Winevt.EVT_FORMAT_MESSAGE_FLAGS#EvtFormatMessageId}, the resource identifier
+     *                          must identify the event's message string.
+     * @param flags             [in] A flag that specifies the message string in the event to format. For possible
+     *                          values, see the {@link Winevt.EVT_FORMAT_MESSAGE_FLAGS} enumeration.
+     * @return A caller-allocated buffer that will receive the formatted message string.
+     */
+    public static Memory EvtFormatMessage(EVT_HANDLE publisherMetadata, EVT_HANDLE event, int messageId, int valueCount,

```

Comment:
I'd return a String - this is a utility method and I'd find that more convenient. See comment for line 47 for a pattern that needs to be applied from line 136 onwards.